src/managers/input/InputManager.cpp

```diff
@@ -774,25 +775,31 @@ void CInputManager::onMouseWheel(IPointer::SAxisEvent e) {
         const auto MOUSECOORDS = g_pInputManager->getMouseCoordsInternal();
         const auto PWINDOW     = g_pCompositor->vectorToWindowUnified(MOUSECOORDS, RESERVED_EXTENTS | INPUT_EXTENTS | ALLOW_FLOATING);
 
-        if (PWINDOW && PWINDOW->checkInputOnDecos(INPUT_TYPE_AXIS, MOUSECOORDS, e))
-            return;
+        if (PWINDOW) {
+            if (PWINDOW->checkInputOnDecos(INPUT_TYPE_AXIS, MOUSECOORDS, e))
+                return;
 
-        if (PWINDOW && *POFFWINDOWAXIS != 1) {
-            const auto BOX = PWINDOW->getWindowMainSurfaceBox();
+            if (*POFFWINDOWAXIS != 1) {
+                const auto BOX = PWINDOW->getWindowMainSurfaceBox();
 
-            if (!BOX.containsPoint(MOUSECOORDS) && !PWINDOW->hasPopupAt(MOUSECOORDS)) {
-                if (*POFFWINDOWAXIS == 0)
-                    return;
+                if (!BOX.containsPoint(MOUSECOORDS) && !PWINDOW->hasPopupAt(MOUSECOORDS)) {
+                    if (*POFFWINDOWAXIS == 0)
+                        return;
 
-                const auto TEMPCURX = std::clamp(MOUSECOORDS.x, BOX.x, BOX.x + BOX.w - 1);
-                const auto TEMPCURY = std::clamp(MOUSECOORDS.y, BOX.y, BOX.y + BOX.h - 1);
+                    const auto TEMPCURX = std::clamp(MOUSECOORDS.x, BOX.x, BOX.x + BOX.w - 1);
+                    const auto TEMPCURY = std::clamp(MOUSECOORDS.y, BOX.y, BOX.y + BOX.h - 1);
 
-                if (*POFFWINDOWAXIS == 3)
-                    g_pCompositor->warpCursorTo({TEMPCURX, TEMPCURY}, true);
+                    if (*POFFWINDOWAXIS == 3)
+                        g_pCompositor->warpCursorTo({TEMPCURX, TEMPCURY}, true);
 
-                g_pSeatManager->sendPointerMotion(e.timeMs, Vector2D{TEMPCURX, TEMPCURY} - BOX.pos());
-                g_pSeatManager->sendPointerFrame();
+                    g_pSeatManager->sendPointerMotion(e.timeMs, Vector2D{TEMPCURX, TEMPCURY} - BOX.pos());
+                    g_pSeatManager->sendPointerFrame();
+                }
             }
+
+            const auto PCURRWINDOW = g_pCompositor->getWindowFromSurface(g_pCompositor->m_pLastFocus.lock());
```

Comment:
I'd generally advise against using m_pLastFocus. It's old, and not very accurate (it has 1001 quirks). Use last pointer focus from the seat manager's current state.